File: go/examples/basic_usage.go
package main

import (
	"encoding/json"
	"fmt"
	"log"

	"github.com/MoyuFunding/exchange-pb/go/pkg/mexc"
	"google.golang.org/protobuf/proto"
)

func main() {
	fmt.Println("=== Exchange PB Go Example ===")

	// Create a book ticker
	bookTicker := &mexc.PublicBookTickerV3Api{
		BidPrice:    "50000.00",
		BidQuantity: "1.5",
		AskPrice:    "50001.00",
		AskQuantity: "2.0",
	}

	fmt.Printf("Bid: %s @ %s\n", bookTicker.BidQuantity, bookTicker.BidPrice)
	fmt.Printf("Ask: %s @ %s\n", bookTicker.AskQuantity, bookTicker.AskPrice)

	// JSON serialization
	jsonData, err := json.MarshalIndent(bookTicker, "", "  ")
	if err != nil {
		log.Fatal("JSON marshal error:", err)
	}
	fmt.Printf("\nJSON representation:\n%s\n", string(jsonData))

	// Protobuf binary serialization
	data, err := proto.Marshal(bookTicker)
	if err != nil {
		log.Fatal("Protobuf marshal error:", err)
	}
	fmt.Printf("\nProtobuf binary size: %d bytes\n", len(data))

	// Deserialize from protobuf binary
	var decoded mexc.PublicBookTickerV3Api
	err = proto.Unmarshal(data, &decoded)
	if err != nil {
		log.Fatal("Protobuf unmarshal error:", err)
	}
	fmt.Printf("Decoded bid price: %s\n", decoded.BidPrice)

	// Create a push data wrapper
	wrapper := &mexc.PushDataV3ApiWrapper{
		Channel:    "spot@public.bookTicker.v3.api",
		Symbol:     "BTCUSDT",
		SymbolId:   "BTC_USDT",
		CreateTime: 1640995200000,
		SendTime:   1640995200001,
		Body: &mexc.PushDataV3ApiWrapper_PublicBookTicker{
			PublicBookTicker: bookTicker,
		},
	}

	fmt.Printf("\nWrapper created for symbol: %s\n", wrapper.Symbol)

	// Serialize wrapper
	wrapperData, err := proto.Marshal(wrapper)
	if err != nil {
		log.Fatal("Wrapper marshal error:", err)
	}
	fmt.Printf("Wrapper binary size: %d bytes\n", len(wrapperData))

	fmt.Println("\n=== Example completed successfully! ===")
}

File: nodejs/examples/basic_usage.js
#!/usr/bin/env node

const { mexc } = require('@moyufunding/exchange-pb');

console.log('=== Exchange PB Node.js Example ===');

// Create a book ticker
const bookTicker = new mexc.PublicBookTickerV3Api({
    bidPrice: "50000.00",
    bidQuantity: "1.5",
    askPrice: "50001.00",
    askQuantity: "2.0"
});

console.log(`Bid: ${bookTicker.bidQuantity} @ ${bookTicker.bidPrice}`);
console.log(`Ask: ${bookTicker.askQuantity} @ ${bookTicker.askPrice}`);

// Convert to plain object for JSON
const plainObject = mexc.PublicBookTickerV3Api.toObject(bookTicker);
console.log('\nJSON representation:');
console.log(JSON.stringify(plainObject, null, 2));

// Protobuf binary serialization
const buffer = mexc.PublicBookTickerV3Api.encode(bookTicker).finish();
console.log(`\nProtobuf binary size: ${buffer.length} bytes`);

// Deserialize from protobuf binary
const decoded = mexc.PublicBookTickerV3Api.decode(buffer);
console.log(`Decoded bid price: ${decoded.bidPrice}`);

// Create a push data wrapper
const wrapper = new mexc.PushDataV3ApiWrapper({
    channel: "spot@public.bookTicker.v3.api",
    symbol: "BTCUSDT",
    symbolId: "BTC_USDT",
    createTime: 1640995200000,
    sendTime: 1640995200001,
    publicBookTicker: bookTicker
});

console.log(`\nWrapper created for symbol: ${wrapper.symbol}`);

// Serialize wrapper
const wrapperBuffer = mexc.PushDataV3ApiWrapper.encode(wrapper).finish();
console.log(`Wrapper binary size: ${wrapperBuffer.length} bytes`);

// Validation example
const error = mexc.PublicBookTickerV3Api.verify(bookTicker);
if (error) {
    console.error('Validation error:', error);
} else {
    console.log('\n✅ Message validation passed');
}

console.log('\n=== Example completed successfully! ==='); 